parsers/IN_UP.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python3
+import re
+from ast import literal_eval
+from operator import itemgetter
+import json
+from arrow import get
+from requests import Session
+import arrow
+from parsers import countrycode
+from parsers.lib import web
+
+
+def fetch_data(country_code, session=None):
+    countrycode.assert_country_code(country_code, 'IN-UP')
+
+    time_now = arrow.now(tz='Asia/Kolkata')
+    india_date = get(time_now, tzinfo='Asia/Kolkata').datetime
+    # india_date = html.find_all('div', 'td-header', recursive=True)
+
+    html_params = {
+        'p_p_id': 'upgenerationsummary_WAR_UPSLDCDynamicDisplayportlet',
+        'p_p_lifecycle': 2,
+        'p_p_state': 'normal',
+        'p_p_mode': 'view',
+        'p_p_resource_id': 'realtimedata',
+        'p_p_cacheability': 'cacheLevelPage',
+        'p_p_col_id': 'column-1',
+        'p_p_col_count': 1,
+        '_upgenerationsummary_WAR_UPSLDCDynamicDisplayportlet_time': time_now,
+        '_upgenerationsummary_WAR_UPSLDCDynamicDisplayportlet_cmd': 'realtimedata'
+    }
+
+    key_map = {
+        'total hydro generation': 'hydro',
+        'total thermal up generation': 'thermal',
+        'cogen-sent out': 'unknown',
+        'solar generation': 'solar',
+        'total up load/demand': 'demand'
+    }
+
+    value_map = {
+        "date": india_date,
```

Comment:
response objects contains several "time_vals" keys, lets use them for datetime rather than arrow.now().